### PR TITLE
Use react dialog for tuned

### PR DESCRIPTION
**Pull request by dperpeet**
The `Dialog` component should be reusable by providing a custom body element and some controlling code, e.g. what happens when the user clicks the primary button.
Notably, the dialog implements the pattern https://github.com/cockpit-project/cockpit/wiki/HIG:-Dialogs with alert/error and in progress states.

### CODE REVIEW

File: pkg/tuned/Makefile.am [hunk 1]
```diff
@@ -1,26 +1,30 @@
+TUNED_PKGDIR=pkg/tuned
+
 tuneddir = $(pkgdatadir)/tuned
 nodist_tuned_DATA = \
-	pkg/tuned/bundle.min.js.gz \
+	$(TUNED_PKGDIR)/bundle.min.js.gz \
```
Comment (stefwalter) [inline pkg/tuned/Makefile.am]: If it's easy, could you include refactoring/renaming in a separate commit? That'll make this easier to review.

Comment (dperpeet) [inline pkg/tuned/Makefile.am]: > If it's easy, could you include refactoring/renaming in a separate commit? That'll make this easier to review.

sure

**Comment by dperpeet** [comment]
based on #4132

**Comment by dperpeet** [comment]
The `show_modal_dialog` function is a convenience function that creates a `<div>` in the DOM and deletes it after the dialog has closed.

**Comment by dperpeet** [comment]
I still need to fix the tests

**Comment by andreasn** [comment]
I had issues getting this to work:
```
Error: Script error for: performance/dialog-view
http://requirejs.org/docs/errors.html#scripterror
 bundle.js:168:17
Error: Script error for: performance/change-profile
http://requirejs.org/docs/errors.html#scripterror
 bundle.js:168:17
Importing XUL nodes into a content document is deprecated. This functionality may be removed soon. jquery-1.7.1.min.js
```

**Comment by andreasn** [comment]
Got it to work now!

**Comment by dperpeet** [comment]
npm fix picked from #4150

**Comment by dperpeet** [comment]
the other commits were merged separately, so this is just the tuned change now

File: pkg/tuned/change-profile.jsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+ *  - click (callback function)
+ */
+var TunedDialogProfile = React.createClass({
+    render: function() {
+        var classes = "list-group-item";
+        if (this.props.selected)
+            classes += " active";
+        var recommended;
+        if (this.props.recommended)
+            recommended = <span className="badge pull-right">recommended</span>;
```
Comment (mvollmer) [inline pkg/tuned/change-profile.jsx]: translatable?

**Comment by mvollmer** [comment]
I like where this is going!

File: pkg/tuned/change-profile.jsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+ *    - description (string)
+ */
+var TunedDialogBody = React.createClass({
+    getInitialState: function() {
+        return {
+            selected_profile: this.props.active_profile,
+        };
+    },
+    componentDidMount: function() {
+        this.props.change_selected(this.props.active_profile);
```
Comment (stefwalter) [inline pkg/tuned/change-profile.jsx]: Why is this line necessary. It doesn't seem to have to do with the DOM. Why use componentDidMount here?

File: pkg/tuned/change-profile.jsx [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    },
+    handleProfileClick: function(profile) {
+        if (profile != this.state.selected_profile) {
+            this.setState( { selected_profile: profile } );
+            this.props.change_selected(profile);
```
Comment (stefwalter) [inline pkg/tuned/change-profile.jsx]: Why do we change the props and the state as a result of this change? Shouldn't it just track in state until the user actually clicks the button? Or are we using a different scheme here?

File: pkg/tuned/dialog-view.jsx [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+        document.addEventListener('keyup', this.keyUpHandler.bind(this));
+    },
+    componentWillUnmount: function() {
+        document.body.classList.remove("modal-in");
+        document.removeEventListener('keyup', this.keyUpHandler.bind(this));
+    },
+    update_progress: function(msg) {
+        this.setState({ action_progress_message: msg });
+    },
+    primary_click: function(e) {
```
Comment (stefwalter) [inline pkg/tuned/dialog-view.jsx]: I think we should tease most of this out. There's should be no concept of a primary button, rather buttons that do things. But perhaps this is work with a separate pull request. The footer should be a separate component and be replacable.

File: pkg/tuned/change-profile.jsx (hunk 2 above)
Comment (dperpeet) [inline pkg/tuned/change-profile.jsx]: using `_("recommended")`, since the `translatable` tag doesn't work correctly when used in js.

**Comment by dperpeet** [comment]
Closed by accident

File: pkg/tuned/change-profile.jsx (hunk 3 above)
Comment (dperpeet) [inline pkg/tuned/change-profile.jsx]: > Why is this line necessary. It doesn't seem to have to do with the DOM. Why use componentDidMount here?

This was to ensure that the controller is initialized to the currently selected value, but it can be omitted.

File: pkg/tuned/change-profile.jsx (hunk 4 above)
Comment (dperpeet) [inline pkg/tuned/change-profile.jsx]: > Why do we change the props and the state as a result of this change? Shouldn't it just track in state until the user actually clicks the button? Or are we using a different scheme here?

As discussed, this doesn't change the props. It updates the state and notifies the controller via callback that the currently selected item has changed.

File: pkg/tuned/dialog-view.jsx (hunk 5 above)
Comment (dperpeet) [inline pkg/tuned/dialog-view.jsx]: > I think we should tease most of this out. There's should be no concept of a primary button, rather buttons that do things. But perhaps this is work with a separate pull request. The footer should be a separate component and be replacable.

Moving the footer out for now, but the concept of different action buttons should be done in a followup.

**Comment by dperpeet** [comment]
rebased
